src/components/Modal/Modal.tsx

```diff
@@ -162,7 +165,7 @@ export const Modal = forwardRef(
       },
     }
 
-    return (
+    const modal = 
       <FocusTrap active={isOpen} focusTrapOptions={focusTrapOptions}>
         <ModalWrapper
           role="dialog"
```

Comment:
Unit tests are looking for role dialog. Do we need to move that attribute onto something wrapping the `modal` used for the portal as well? 

If we do, I like the pattern in [here ](https://assortment.io/posts/accessible-modal-component-react-portals-part-2)where its a default but can be overridden. This could be a good chance to surface props for that role attribute which might change depending on use case.  In the new code, seems like the role goes on the modal wrapper when `renderToPortal` is false;  otherwise the role goes on `modal` itself ... _I think_

Reply:
I think the issue was that because the `Modal` was being directly rendered into `document.body`, it wasn't within the `div` that React Testing Library was looking inside of when `screen.getByRole` was called. I fixed this in the next commit and modified the behavior to use a div with an `id` of `modal-root` to render modals if it exists.

Reply:
ah makes sense! Thanks for explaining